azure-pipelines.yml

```diff
@@ -1,49 +1,36 @@
 jobs:
-- job: core
+
+- job: azure_pipeline_tests
   steps:
+
+  # Run a minimal set of tests in normal Linux pipeline builds
   - task: Gradle@2
-    displayName: Build & test
+    condition: eq(variables['Agent.OS'], 'Linux')
+    displayName: Build & test (Linux - minimal core)
     env:
       AWS_ACCESS_KEY_ID: $(aws.accessKeyId)
       AWS_SECRET_ACCESS_KEY: $(aws.secretAccessKey)
     inputs:
         gradleWrapperFile: 'gradlew'
         jdkVersionOption: '1.11'
-        options: '--no-daemon --continue'
-        tasks: 'testcontainers:check'
+        options: "--no-daemon --continue"
+        tasks: "testcontainers:test --tests GenericContainerRuleTest"
```

Comment:
IIRC we can change it based on conditions (all on Windows, subset otherwise)

Reply:
I thought so too, and have been scouring the documentation to try and find out how to do this, but so far no luck. Have you seen anything concrete?

Reply:
try https://docs.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#conditional-insertion

Reply:
Aha, I'd missed that - cheers.

Reply:
The `Agent.OS` variable is not set at the point in time when conditional insertion takes place, so I was unable to set a variable.

For example:
```
variables:
  ${{ if eq(variables['Agent.OS'], 'Linux') }}:
    # Run minimal core subset of tests on Linux
    tasks: "testcontainers:test --tests GenericContainerRuleTest"
  ${{ if eq(variables['Agent.OS'], 'Windows_NT') }}:
    # Run all checks on Windows
    tasks: "check"
```

would effectively do nothing, because neither branch is true. The `tasks` variable was unset at the time of use in the `Gradle@2` task.

Just in case something else was the problem, I tried:

```
variables:
  ${{ if eq(variables['Agent.OS'], 'Linux') }}:
    # Run minimal core subset of tests on Linux
    tasks: "testcontainers:test --tests GenericContainerRuleTest"
  ${{ if ne(variables['Agent.OS'], 'Windows_NT') }}:
    # Run all checks on Windows
    tasks: "check"
```

(the difference is `ne`, not `eq`, on the second branch).

With this, the `tasks` variable _was set_, but always to `"check"`. So this demonstrates that the conditional insertion is working, and the placeholder resolution is working - the problem is just that `Agent.OS` is not set when the pipeline file is compiled, which is before it hits a build agent.

I rolled back the branch to get rid of the messy trail of commits that I created while trying to figure this out! 😅